### ISSUE TITLE
IDE0037 is not working

### ISSUE THREAD

**Issue by mpagliaro98** [dotnet/roslyn#75428]
**Version Used**:

Visual Studio Enterprise 2022 17.9.6

**Steps to Reproduce**:

A minimal reproduction of this issue is attached as a .NET 8 C# class library project in a solution with an editorconfig file. The editorconfig was generated from the "Generate .editorconfig file from settings" button in Tools > Options > Text Editor > C# > Code Style section in Visual Studio, with only the changes described in this post made to it.

I am trying to build an editorconfig to share consistent code style guidelines with my organization, but I noticed some rules are not enforced at all. These lines are in my editorconfig, indicating I want all tuples and anonymous types to have explicit names shown:
```
dotnet_style_prefer_inferred_anonymous_type_member_names = false
dotnet_style_prefer_inferred_tuple_names = false
```

In Visual Studio's built-in editorconfig editor, I've also set these two to warning (though I'm not sure where this severity gets set, it doesn't get set in the editorconfig file):
![image](https://github.com/user-attachments/assets/55da0c86-60d9-4ef2-9266-c2755db7adeb)

I've written this basic code to test if this rule is being applied. Going off of [official Microsoft documentation](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0037), the test1 and test3 lines should be showing a warning, but when I'm in this solution, none of these lines show any diagnostic for IDE0037.
```
        int age = 41;
        string name = "john";

        // prefer inferred tuple element names: No
        // this line should be a warning
        var test1 = (age, name);
        // this line should be allowed
        var test2 = (PersonAge: age, ContactName: name);
        // but there is no diagnostic on either

        // prefer inferred anonymous type member names: No
        // this line should be a warning
        var test3 = new { age, name };
        // this line should be allowed
        var test4 = new { AnonAge = age, AnonName = name };
        // but there is no diagnostic on either
```

I've also set this line manually in my editorconfig, but the warning is still not showing, neither in the editor nor on build:
`dotnet_diagnostic.IDE0037.severity = warning`

These settings can also be set in Visual Studio for just my workstation, but this doesn't seem to work either. In Tools > Options > Text Editor > C# > Code Style, the same settings for prefer inferred tuple element names and prefer inferred anonymous type member names show. When I change it from "yes" to "no", the example code below is supposed to update to swap which is preferred, but the code does not change when I set it to "no". The editorconfig file should be overriding these settings anyway.
![image](https://github.com/user-attachments/assets/6ef4c5d1-9bea-46a8-be7c-cdbb50835860)

I also suspect more of these options are not working, especially those in the "expression preferences" section of the dialog pictured above. A few others I checked also did not change the example code when switching between "yes" and "no", such as "prefer explicit tuple name". I tried to get this one to work with an editorconfig as well and didn't get it to work before making this repro to test with.

**Diagnostic Id**:

IDE0037: Use inferred member names

**Expected Behavior**:

Using the configuration described above, IDE0037 would display a warning on the described lines of code

**Actual Behavior**:

No warning or any diagnostic message is displayed

**Comment by mpagliaro98** [comment]
[IDE0037 bug.zip](https://github.com/user-attachments/files/17286833/IDE0037.bug.zip)

My reproduction didn't attach to the main message... here it is!

**Comment by CyrusNajmabadi** [comment]
> indicating I want all tuples and anonymous types to have explicit names shown

I don't believe we have a setting for that. We only have the reverse, forcing implicit naming. You can either force in that direction, or turn the rule off.

We have very few rules which are tri-state (one way, or the other, or off). 

I'd recommend you write your own analyzer if you want to force this on your code base.

**Comment by mpagliaro98** [comment]
Just did some more testing and noticed Visual Studio was adding a large block of duplicate rules to the bottom of my editorconfig file with the header `[*.{cs,vb}]`. I tried to remove this block, but any time I used the built-in editorconfig editor in visual studio to change a property, it would add it back in. The settings in this block were overriding several things I was trying to set manually further up, making it seem like several features were not working. I'll close this issue since it seems like there isn't actually an issue here.

Thanks for the insight about the lack of tristate logic in the built-in analyzers! I'll be able to write some simple analyzers for my team to enforce those code quality rules.
